packages/react-components/src/components/Textarea/Textarea.spec.tsx

```diff
@@ -2,59 +2,69 @@ import * as React from 'react';
 
 import { vi } from 'vitest';
 
-import { render } from 'test-utils';
+import { render, userEvent } from 'test-utils';
 
 import noop from '../../utils/noop';
 
 import { Textarea, TextareaProps } from './Textarea';
 
-import styles from './Textarea.module.scss';
-
-const baseClass = 'textarea';
-
 const renderComponent = (props: Partial<TextareaProps>) => {
   const finalProps = Object.assign(
     {},
     { onChange: noop },
     props
   ) as TextareaProps;
 
-  return render(<Textarea {...finalProps} className="my-css-class" />);
+  return render(<Textarea {...finalProps} />);
 };
 
 describe('<Textarea> component', () => {
   it('should allow for custom class', () => {
-    const { container } = renderComponent({});
+    const { container } = renderComponent({ className: 'my-css-class' });
 
     expect(container.firstChild).toHaveClass('my-css-class');
   });
 
-  it('should have disabled class on disabled', () => {
-    const { container } = renderComponent({ disabled: true });
-
-    expect(container.firstChild).toHaveClass(styles[`${baseClass}--disabled`]);
-  });
-
-  it('should have error class on error', () => {
-    const { container } = renderComponent({ error: true });
+  it('should be disabled if prop is given', () => {
+    const { getByRole } = renderComponent({ disabled: true });
 
-    expect(container.firstChild).toHaveClass(styles[`${baseClass}--error`]);
+    expect(getByRole('textbox')).toBeDisabled();
   });
 
   it('should call custom onFocus and onBlur', () => {
     const customOnFocus = vi.fn();
     const customOnBlur = vi.fn();
-    const { container } = renderComponent({
+    const { getByRole } = renderComponent({
       onFocus: customOnFocus,
       onBlur: customOnBlur,
     });
 
-    const textarea = container.querySelector('textarea');
-    textarea?.focus();
-
+    userEvent.click(getByRole('textbox'));
     expect(customOnFocus).toHaveBeenCalled();
 
-    textarea?.blur();
+    userEvent.click(document.body);
     expect(customOnBlur).toHaveBeenCalled();
   });
+
+  it('should call onChange if textarea value change', () => {
```

Comment:
I'm not sure we need this test, because onChange callback is handled by a native implementation